lib/lte_link_control/lte_lc.c

```diff
@@ -880,60 +877,37 @@ int lte_lc_psm_get(int *tau, int *active_time)
 	 * Periodic-TAU.
 	 */
 
-	response[0] = '\0';
+	err = nrf_modem_at_scanf(
+		"AT%XMONITOR",
+		"%%XMONITOR: "
+		"%*u,"          /* <reg_status> */
+		"%*[^,],"       /* <full_name> */
+		"%*[^,],"       /* <short_name> */
+		"%*[^,],"       /* <plmn> */
+		"%*[^,],"       /* <tac> */
+		"%*d,"          /* <AcT> */
```

Comment:
They are `%d` or `%u` based on what type of parameter would be used for the in `struct lte_lc_cell` if they were not ignored here.
I didn't think about this really bu used the version from MOSH's `link_api.c` which stores most of the parameters.

Reply:
`AcT` is not present in `struct lte_lc_cell`. It has own enum `lte_lc_lte_mode`, which can have values `0`, `7` or `9`.

`rsrp` and `snr` are encoded differently in `%XMONITOR` notification and in `struct lte_lc_cell`. In the AT notification, the value ranges are `0...97, 255` and `0...49, 127`, respectively.

Reply:
I can change them whatever way we want. But I would but `%d` instead of `%u` for all if one is selected to be more robust.

Reply:
I agree that this is quite theoretical, but if the `%XMONITOR` notification would contain a negative value for one of the fields, I think it would be better to fail in parsing than silently accept the unexpected value. I disagree that accepting a negative value would make it more robust, because that would result in undefined behavior after the parsing. 😃

Reply:
> I agree that this is quite theoretical, but if the `%XMONITOR` notification would contain a negative value for one of the fields, I think it would be better to fail in parsing than silently accept the unexpected value. 

My thinking is that letting it go through here is fine because the error (if it would occur) might be hard to spot given scanf parsing hides many issues which doesn't produce meaningful error. In practice the return value from `nrf_modem_at_scanf` would just be 0 meaning no interested arguments from the end of the 

> I disagree that accepting a negative value would make it more robust, because that would result in undefined behavior after the parsing. 😃

We are ignoring all integer parameters here so it would not be failing afterwards or causing undefined behavior with this given AT response.

Reply:
It seems `%u` doesn't fail when you have `-1000` so any number is stored irrespective of `%d` or `%u` and the type of the argument where the value is stored.
Using `%u` for all integer fields.